openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -316,7 +317,7 @@ def require_permission_for_library_key(library_key, user, permission):
         raise PermissionDenied
 
 
-def get_library(library_key):
+def get_library(library_key, user=None):
     """
     Get the library with the specified key. Does not check permissions.
```

Comment:
You can see here, this says "Does not check permissions."
So we need to either update this docstring or change the approach. I think I would prefer a slightly change to the approach, and move the permissions checking into the REST API. Right now all of the code in `content_libraries` follows this pattern: the code in `api` doesn't bother with permissions at all, and the code in `views.py` does.

Reply:
Updated: https://github.com/openedx/edx-platform/pull/34934/commits/c303069324b444647ac3006fee882ba42d7f42e7